### DISCUSSION TITLE
Which Version for which Chip/Module? esp: "ESP32-WROOM32"

### DISCUSSION THREAD

**Discussion by reimerp** [arendst/Tasmota#22252] [Q&A] [answered]
I have a ESP32-D0WDQ6 (revision 1) on an Wroom32 DevBoard. ChipVersion was found by esptool.
But which tasmota version should I use -- or which platformio board?
This is prob documented somewhere but where?

**Comment by sfromis** [comment]
The Tasmota binaries have names matching the hardware platform. In your case staring with `tasmota32`, but there's also `tasmota32s2`, `tasmota32s3`, `tasmota32c2`, `tasmota32c3`, `tasmota32c6` and just `tasmota` for ESP8266. This "happens" to align pretty well with `--chip` selection in `esptool.py`, but the simplest is to not even use this tool.

Using the Tasmota Web Installer https://tasmota.github.io/install/ can automatically select the binary to match the hardware platform, which importantly includes using the `.factory` variant of ESP32 binaries for initial wired flashing.

**Comment by reimerp** [comment]
There are hints on firmware/template selection on [blackadders fantastic pages](https://templates.blakadder.com/) for tons of knowns modules. but there are so many chips variants available combined to modules with additional components, resulting in a zoo of varieties. Some are clearly marked (I have a C3 as well) but some are not listed, Web Installer is a great help for beginners but    cannot cover this. Still looking for a "**WROOM32**" Module ist just the bare module attached to a seperate available breakout board.

**Reply by sfromis** [reply]
I just told you that you need the `tasmota32` variant "in your case", without the "suffix" for newer iterations. This has **nothing** to do with it being on a development board, or just a "naked" module, but is meant to support he smaller chip (about 8x8 mm) independently of what board it is soldered to (often inside a metal shield).

And that's why the Tasmota Web Installer perfectly covers this, as long as you've got to the point of getting it in flashing/bootloader mode, which can vary depending on board design and ESP32 variant. If you are handling the "naked" ESP32-WROOM modules (around 2.5 x 1.8 cm) without them being on a larger board (like a development board), you are expected to read data sheets about how to handle the modules. Still, development boards (around 5 x 2.5 cm and up) will typically have an USB socket and adapter provided to make this easy. 

What you get in the templates repository is focused on a different thing, having pin assignments for a long list of commercial devices, making it easier to get Tamtoa running on some plug or lamp.

**Comment by Jason2866** [comment, marked as answer]
Short answer. For every esp32 chip whatever module or naming it has it is tasmota32.
